### PR TITLE
Add nightly running of golden and trace tests.

### LINKED ISSUE #200: Add nightly golden and trace test runs on master branch
Ensure the CI is setup to run these and surface any failures.  Determine suitable off-peak time.  See if its feasible to run multi-node, single actor trace tests as part of this.

### PR DESCRIPTION
**Pull request by scttl**
Resolves #200

### CODE REVIEW

**Comment by scttl** [comment]
Unfortunately these new nightly tests won't actually start executing until this code is merged but I don't foresee any issues (syntax already ok given other test runs, not running any net new job types).

File: .circleci/config.yml [hunk 1]
```diff
@@ -378,3 +378,20 @@ workflows:
...
+            branches:
+              only:
+                - master
+    jobs:
+      - unit_tests
+      - integration_tests
+      - golden_test_gym
+      - golden_test_doom
+      - golden_test_mujoco
+      - trace_test_gym
```
Comment (balajismaniam) [inline .circleci/config.yml]: We can remove or comment the trace tests for now as they are going to fail now. Otherwise LGTM.

Comment (scttl) [reply .circleci/config.yml]: yeah good point.  Will push an update.

**Comment by balajismaniam** [comment]
This is also 1am UTC. I don't think it matters but we can change it if necessary.

**Comment by scttl** [comment]
yeah I was trying to find a relatively low usage 3+ hour sweetspot between Israel local dev time (3am) and west-coast US dev time (5pm)

**Pull request merged**